src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/service/MergeAlertService.kt

```diff
@@ -107,13 +78,47 @@ class MergeAlertService(
         ),
       )
       alert
+    }
+
+    var domainEventRegistered = false
+    val domainEvent = AlertsMergedEvent(
+      prisonNumberMergeFrom = request.prisonNumberMergeFrom,
+      prisonNumberMergeTo = request.prisonNumberMergeTo,
+      mergedAlerts = alertsCreated,
+    )
+
+    newAlerts.map {
+      if (!domainEventRegistered) {
+        it.registerAlertsMergedEvent(domainEvent)
+        domainEventRegistered = true
+      }
+      alertRepository.save(it)
     }.also {
       it.logDuplicateActiveAlerts(request.prisonNumberMergeTo)
       alertRepository.flush()
     }
 
+    val alertsDeleted = alertRepository.saveAllAndFlush(
```

Comment:
Is it worth adding a bit of doc here explaining why we're having to add the event here rather than in its own entity, as you explained in slack?